FBSnapshotTestCase/FBSnapshotTestCase.h

```diff
@@ -28,11 +24,13 @@
  */
 #define FBSnapshotVerifyViewWithOptions(view__, identifier__, suffixes__) \
 { \
+NSString *envReferenceImageDirectory = [NSProcessInfo processInfo].environment[@"FB_REFERENCE_IMAGE_DIR"]; \
```

Comment:
Well, the thing is that in a mixed test codebase as @xing will have it we would need to set both (`-D` flag and Environment variable). So checking if `-D FB_REFERENCE_IMAGE_DIR` flag exists will succeed in Swift but will have the wrong value.
For me the best solution is to make a breaking change. This has 3 benefits:

* it works 3rd party matchers like [Nimble-Snapshots](https://github.com/ashfurrow/Nimble-Snapshots) and [Expecta matchers](https://github.com/dblock/ios-snapshot-test-case-expecta) and avoids [workarounds like](https://github.com/dblock/ios-snapshot-test-case-expecta#sane-defaults)
```obj-c
beforeAll(^{
    setGlobalReferenceImageDir(FB_REFERENCE_IMAGE_DIR);
});
```
* it is uniform for Objective-C and Swift so switching projects is easier
* code does not differentiate if running in Swift or Objective-C that ensures maintainability